src/rgw/rgw_kms.cc

```diff
@@ -282,10 +282,6 @@ class VaultSecretEngine: public SecretEngine {
     }
 
     secret_req.append_header("X-Vault-Token", vault_token);
-    if (!vault_token.empty()){
-      secret_req.append_header("X-Vault-Token", vault_token);
-      vault_token.replace(0, vault_token.length(), vault_token.length(), '\000');
```

Comment:
I can't find any reference in Vault documentation that it would ever want an empty X-Vault-Token header, do you know of any such? In that case it would make more sense to not add it when it's empty and keep the branch.

It also zeroes out the memory for the vault_token stored in memory so we should not remove that, even though it's a simple security measure I don't think we should lower the bar.